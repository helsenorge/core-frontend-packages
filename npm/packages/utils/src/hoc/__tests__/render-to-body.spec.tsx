import * as React from 'react';

import { mount } from 'enzyme';

import RenderToBody from '../render-to-body';

describe('render-to-body', () => {
  describe('Gitt at RenderToBody wrapper en vanlig Element', () => {
    describe('N책r printable er false', () => {
      it('S책 renderes det en portal', () => {
        const wrapper = mount(
          <RenderToBody>
            <section>{'My first section rendered to document'}</section>
          </RenderToBody>
        );

        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('N책r printable er true', () => {
      it('S책 renderes det en portal med body tag', () => {
        const wrapper = mount(
          <RenderToBody printable>
            <section>{'My second section rendered to body (printable)'}</section>
          </RenderToBody>
        );

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
