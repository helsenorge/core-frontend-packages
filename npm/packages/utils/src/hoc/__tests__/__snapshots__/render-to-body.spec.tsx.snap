// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render-to-body Gitt at RenderToBody wrapper en vanlig Element N책r printable er false S책 renderes det en portal 1`] = `
<RenderToBody>
  <Portal
    containerInfo={
      <div>
        <section>
          My first section rendered to document
        </section>
      </div>
    }
  >
    <section>
      My first section rendered to document
    </section>
  </Portal>
</RenderToBody>
`;

exports[`render-to-body Gitt at RenderToBody wrapper en vanlig Element N책r printable er true S책 renderes det en portal med body tag 1`] = `
<RenderToBody
  printable={true}
>
  <Portal
    containerInfo={
      <body>
        <div>
          <section>
            My first section rendered to document
          </section>
        </div>
        <section>
          My second section rendered to body (printable)
        </section>
      </body>
    }
  >
    <section>
      My second section rendered to body (printable)
    </section>
  </Portal>
</RenderToBody>
`;
